test/ConfigAggregatorTest.php

```diff
@@ -127,46 +121,18 @@ function () {
         $this->assertEquals(0600, fileperms($this->cacheFile) & 0777);
     }
 
-    public function testConfigAggregatorSetsHandlesUnwritableCache()
+    public function testConfigAggregatorThrowsExceptionOnUnwritableCache()
     {
+        $this->expectException(RenameException::class);
         chmod(dirname($this->cacheFile), 0400);
-
-        $foo = function () {
+        $suppressNotice = function () {
             new ConfigAggregator([
                 function () {
                     return ['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true];
                 }
             ], $this->cacheFile);
         };
-        @$foo(); // suppress warning
-
-        $errors = error_get_last();
-        $this->assertNotNull($errors);
-        $this->assertFalse(file_exists($this->cacheFile));
-    }
-
-    public function testConfigAggregatorRespectsCacheLock()
-    {
-        $expected = [
-            'cache' => 'locked',
-            ConfigAggregator::ENABLE_CACHE => true,
-        ];
-
-        $fh = fopen($this->lockFile, 'c');
-        flock($fh, LOCK_EX);
-        file_put_contents($this->cacheFile, '<' . '?php return ' . var_export($expected, true) . ';');
-
-        $method = new ReflectionMethod(ConfigAggregator::class, 'cacheConfig');
-        $method->setAccessible(true);
-        $method->invoke(
-            new ConfigAggregator(),
-            ['foo' => 'bar', ConfigAggregator::ENABLE_CACHE => true],
-            $this->cacheFile
-        );
-        flock($fh, LOCK_UN);
-        fclose($fh);
-
-        $this->assertEquals($expected, require $this->cacheFile);
+        @$suppressNotice(); // suppress "file created in the system's temporary directory"
```

Comment:
I really don't like this. I would prefer if this gets resolved without error suppression.

@michalbundyra do you think safe writer should intercept and suppress that `tempname()` warning?

Reply:
@Xerkus Can you propose an alternative? Do you think that webimpress/safe-writer should handle this? And, if so... you realize it would have to be done similarly, right? Would you be okay with it in the library we consume, instead of here?

(I just want to try and unblock this. Saying a preference without providing some options can block progress.)

Reply:
@weierophinney I believe warning here is not because of a fallback to temporary location but because we use folder inside temporary directory: `$dir = sys_get_temp_dir() . '/mezzio_config_loader';`
I think it might be valid to intercept and silence that warning in safe writer for this case. But only that warning.

Here we suppress anything, including any potential problems.
I see two solutions:
- Use custom error handler. Suppress this very specific error
- Move target directory to gitignored test asset folder within `test/`

Reply:
@Xerkus Thanks - that's exactly the sort of guidance I was hoping for!

Reply:
@Xerkus @weierophinney `safe-writer` tries to create a temp file in the same directory as the config. The test is making that read-only, hence the notice when it falls back to the system temp dir.

I don't like the suppression either, but I kind of think it _is_ a safe-writer problem: what if the temp dir is on a different filesystem to the target file? Will the `rename()` still be atomic?

Looking at it again, the test used to check the cache file wasn't created. Now it's checking if an exception was thrown - which could be a BC break. Grrr...

Reply:
@Xerkus @weierophinney OK, best of both worlds approach. I've submitted a [PR](https://github.com/webimpress/safe-writer/pull/6) upstream to handle this, and added an error handler here until that's merged.

I've also reverted to the previous behaviour, where errors saving the cache were silently ignored. That's probably not what's wanted long term, but prevents the BC break.